### PR TITLE
Replace Flake8 with Ruff for linting and automatic fixes

### PR DESCRIPTION
**Pull request by taylorbell57**
## Summary

Replace Eureka!'s existing Flake8 linting with Ruff while preserving the current Pycodestyle and Pyflakes rule coverage. Ruff enables pre-commit.ci to apply available safe lint fixes automatically (saving us some manual labor).

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Code cleanup / restructure
- [ ] Documentation
- [x] Other (describe in "What changed")

## Stages affected

- [ ] S1 — Detector processing
- [ ] S2 — Calibrations
- [ ] S3 — Data reduction
- [ ] S4 — Lightcurve generation
- [ ] S5 — Lightcurve fitting
- [ ] S6 — Planet spectra
- [ ] lib / utilities
- [x] CI/CD / GitHub Actions

## What changed

- Added Ruff configuration matching the existing Flake8 rules, exclusions, and per-file ignores
- Enabled Ruff preview rules required for equivalent Pycodestyle coverage
- Replaced the Flake8 GitHub Actions workflow with Ruff
- Added `ruff check --fix` to pre-commit so pre-commit.ci can automatically commit safe fixes
- Replaced the Flake8 dependency with Ruff in `pyproject.toml`, `environment.yml`, and the oldest-practical constraints
- Updated the contributor documentation and README status badge
- Removed the obsolete `.flake8` configuration
- Resolved two Ruff/Flake8 edge-case differences without changing runtime behavior

## Related issues

N/A

## Testing

- [x] I ran `pytest` locally and all relevant tests pass

**Test notes** (what was tested; if any tests are failing, explain why):

No runtime behavior was changed, so the full pytest suite was not run. The following checks passed:

- `pre-commit run --all-files`
- Repository-wide Ruff linting
- YAML and TOML validation
- isort
- `git diff --check`

## Other Notes

Ruff applies every available safe fix automatically. Violations without an automatic fix will continue to fail CI and require manual correction.

### CODE REVIEW

File: tests/test_MIRI.py [hunk 1]
```diff
@@ -82,7 +79,10 @@ def test_MIRI(capsys, keep_s2_output, keep_s3_output, keep_s4_output):
     # run assertions for S2
     if s2_installed:
         # Only run S1-2 stuff if jwst package has been installed
-        # meta.outputdir_raw=f'{os.sep}data{os.sep}JWST-Sim{os.sep}MIRI{os.sep}Stage1{os.sep}'
+        # meta.outputdir_raw = (
+        #     f'{os.sep}data{os.sep}JWST-Sim{os.sep}MIRI'
+        #     f'{os.sep}Stage1{os.sep}'
+        # )
```
Comment (kevin218) [inline tests/test_MIRI.py:85]: Should we just delete all of the commented out S1-related code in this test or do you want to keep it for later?

Comment (taylorbell57) [reply tests/test_MIRI.py:85]: Part of me had hoped to add some S1 coverage at some point. Now that S1 runs so much faster and with so much less RAM, it's as feasible as it has ever been, but I'm not sure it's feasible enough
